modules/caddyhttp/responsewriter.go

```diff
@@ -122,46 +121,34 @@ type responseRecorder struct {
 //     }
 //     // process the buffered response here
 //
-// After a response has been buffered, remember that any upstream header
-// manipulations are only manifest in the recorder's Header(), not the
-// Header() of the underlying ResponseWriter. Thus if you wish to inspect
-// or change response headers, you either need to use rec.Header(), or
-// copy rec.Header() into w.Header() first (see caddyhttp.CopyHeader).
+// The header map is not buffered; i.e. the ResponseRecorder's Header()
+// method returns the same header map of the underlying ResponseWriter.
+// This is a crucial design decision to allow HTTP trailers to be
+// flushed properly (https://github.com/caddyserver/caddy/issues/3236).
```

Comment:
I'm a little unclear on how this works - does this mean the headers can't be modified (i.e. by `header` or `header_down`) after we get the response from the proxied service? I'm not sure I see the big picture of when that 
would happen.

Reply:
(To be clear, this particular commit was already cherry-picked into master.)

I believe this change does not affect the manipulation of headers, it only eliminates the map-in-the-middle so to speak.